core/src/main/java/org/elasticsearch/action/bulk/BulkProcessor.java

```diff
@@ -348,7 +348,7 @@ public void onFailure(Throwable e) {
                 });
                 success = true;
             } catch (InterruptedException e) {
-                Thread.interrupted();
+                Thread.currentThread().interrupt();
```

Comment:
We've discussed that this will go into a separate PR as it directly affects the client and I need to check anyway whether we're handling all cases in `BulkProcessor` consistently (i.e. numberOfConcurrentRequest > 0 vs. numberOfConcurrentRequest == 0)